include/aspect/simulator_access.h

```diff
@@ -901,6 +902,74 @@ namespace aspect
       Particle::World<dim> &
       get_particle_world();
 
+      /**
+       *  Return true if using the block GMG Stokes solver.
+       */
+      bool is_matrix_free();
```

Comment:
we currently do not solve temperature equations matrix-free but we might at some point, so this should contain ``stokes`` somewhere in the name.